### PR TITLE
Vhostuser network enabling

### PR DESCRIPTION
**Pull request by egernst**
PR to enable VPP for POC work on 01org/cc-oci-runtime's network/vhost-user-poc branch.

OVS plugin should be updated to make use of this branch instead of OVS-DPDK POC going forward.

Verified with latest cc-oci-runtime released, 2.1.4.

### CODE REVIEW

**Comment by mcastelino** [comment]
@egernst can we squash the commits. Here most of the commits are around documentation and intermediate bug fixes.

File: data/hypervisor.args.in [hunk 1]
```diff
@@ -2,7 +2,7 @@
 -name
 @NAME@
 -machine
-pc-lite,accel=kvm,kernel_irqchip,nvdimm
+pc,accel=kvm,kernel_irqchip,nvdimm
```
Comment (mcastelino) [inline data/hypervisor.args.in]: @egernst we should call this out somewhere in the documentation that we are currently using pc vs pc-lite.

File: configure.ac [hunk 2]
```diff
@@ -302,7 +302,7 @@ AS_IF([test x"$with_tests_bundle_path" != x""],
       [])
 
 #Kernel path for Clear Containers
-DEFAULT_KERNEL_PATH=$localstatedir/lib/$PACKAGE_NAME/data/kernel/vmlinux.container
+DEFAULT_KERNEL_PATH=$localstatedir/lib/$PACKAGE_NAME/data/kernel/vmlinuz.container
```
Comment (mcastelino) [inline configure.ac]: @egernst do we install this by default in the case of obs packages.

File: documentation/Using-VPP-and-COR.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+## Example: Launching two Clear Containers using VPP
+
+To make use of VPP, the next step is to use Docker to create a network using the OVS-DPDK switch.  An example:
+```
+$ sudo docker network create -d=vpp --ipam-driver=vpp --subnet=192.168.1.0/24 --gateway=192.168.1.1  vpp_net
+```
+
+You can now test connectivity by launching two containers as follows (this assumes that you have Docker setup to use this branch's runtime):
+```
+$ sudo docker run --net=vpp_net --ip=192.168.1.2 --mac-address=CA:FE:CA:FE:01:02 -it debian bash -c "ip a; ip route; sleep 300"
```
Comment (mcastelino) [inline documentation/Using-VPP-and-COR.md]: @egernst we can switch to the new -itd and sh at the end vs the sleep.

File: src/networking.h [hunk 4]
```diff
@@ -23,6 +23,12 @@
 
 #include "netlink.h"
 
+/* openvswitch will place the vhost-user socket in a particular location,
+ * and we chose to name the interface ovd_<ip address> during endpoint creation
+ * in the OVSDPDK Docker plugin.  OVS_PORT_PATH is used to identify this file.
+ */
+#define OVS_PORT_PATH "/tmp/v_%s"
```
Comment (mcastelino) [inline src/networking.h]: @egernst did we change this from /var/run/...

File: configure.ac (hunk 2 above)
Comment (egernst) [inline configure.ac]: Yes - see https://github.com/01org/cc-oci-runtime/issues/816

File: src/networking.h (hunk 4 above)
Comment (egernst) [inline src/networking.h]: right; it was an OVS specific path before.

Comment (egernst) [inline src/networking.h]: I'll need to update that naming, as this isn't OVS specific path any longer....

Comment (mcastelino) [inline src/networking.h]: ah ok. So unified handling for both dpdk and VPP. Then what is the role of the /tmp/v_file

"The runtime determines that a vhost-user enbaled interface is desired if
the IP for a discovered network interface matches the address appended to a
/tmp/v_ file."

Also there are a couple of typos in the commit message

**Comment by mcastelino** [comment]
LGTM

**Comment by rcaballeromx** [comment]
@egernst: I have edited the VPP file, please take a look. However, the OVS file does not reflect the edits I already did. Should I apply those edits anew? Please let me know if you prefer this method over me commenting all the needed changes here.

**Comment by egernst** [comment]
That's a great way to do it @rcaballeromx  -- thanks.  The existing OVS should match what we went through a couple weeks ago.  If you see any issues, no problem editing in the same way you did for VPP on this PR

**Pull request merged**